cds_ils/importer/eitems/importer.py

```diff
@@ -73,15 +79,19 @@ def _set_record_import_source(self, record_dict):
 
     def _eitem_has_higher_priority(self, eitem, priority=None):
         """Replace the eitems with higher priority providers."""
-        eitem_provider = self._get_record_import_provider(eitem)
-        if not priority:
-            priority = self.current_provider_priority
+        priority = priority or self.current_provider_priority
 
         PRIORITY_FOR_UNKNOWN = 100  # high number means low priority
-        providers = current_app.config["CDS_ILS_IMPORTER_PROVIDERS"]
-        eitem_priority = providers.get(eitem_provider, {}).get(
-            "priority", PRIORITY_FOR_UNKNOWN
-        )
+        PROVIDERS = current_app.config["CDS_ILS_IMPORTER_PROVIDERS"]
+
+        eitem_provider = self._get_record_import_provider(eitem)
+        if eitem_provider:
+            provider_definition = PROVIDERS.get(eitem_provider, {})
+            eitem_priority = provider_definition.get(
+                "priority", PRIORITY_FOR_UNKNOWN
+            )
+        else:
+            eitem_priority = PRIORITY_FOR_UNKNOWN
```

Comment:
```suggestion
       eitem_priority = PRIORITY_FOR_UNKNOWN
       
       provider_definition = PROVIDERS.get(eitem_provider, {})
       eitem_priority = provider_definition.get(
          "priority", PRIORITY_FOR_UNKNOWN
      )
```

Reply:
OK fixed, I was not super happy to potentially search in `PROVIDERS` for a `None` or empty value, but I am ok with that.